cmd/collector/app/sanitizer_v2/utf8_sanitizer.go

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package sanitizer_v2
+
+import (
+	"fmt"
+	"unicode/utf8"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+const (
+	invalidOperation = "InvalidOperationName"
+	invalidService   = "InvalidServiceName"
+	invalidTagKey    = "InvalidTagKey"
+)
+
+// UTF8Sanitizer sanitizes all strings in spans.
+type UTF8Sanitizer struct {
+	logger *zap.Logger
+}
+
+// NewUTF8Sanitizer creates a UTF8 sanitizer with logging functionality.
+func NewUTF8Sanitizer(logger *zap.Logger) SanitizeTraces {
+	return UTF8Sanitizer{logger: logger}.Sanitize
+}
+
+// Sanitize sanitizes the UTF8 in the spans.
+func (s UTF8Sanitizer) Sanitize(traces ptrace.Traces) ptrace.Traces {
+	resourceSpans := traces.ResourceSpans()
+	for i := 0; i < resourceSpans.Len(); i++ {
+		resourceSpan := resourceSpans.At(i)
+		scopeSpans := resourceSpan.ScopeSpans()
+		for j := 0; j < scopeSpans.Len(); j++ {
+			spans := scopeSpans.At(j).Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				if !utf8.ValidString(span.Name()) {
+					s.logger.Info("Invalid utf8 operation name", zap.String("operation_name", span.Name()))
```

Comment:
I don't think we should be logging, let's keep sanitizers dependency-free.

Generally we want sanitizers to preserve the original information. It's fine to replace the operation_name, but we can also store the original value (same for all other attributes) as a new attribute with `bad_udf8_` prefix and record the value as binary, instead of string. This keeps the visibility to the original malformed data without polluting the operational logs.